src/cpp_generator.cpp

```diff
@@ -207,9 +207,22 @@ cpp_generator::function cpp_generator::generate_module(const module& m,
     });
     f.set_name(name).set_types(m).set_body(
         m, [&](instruction_ref ins, const auto& names) -> std::string {
-            if(ins->name() == "@literal")
-                return shape::cpp_type(ins->get_shape().type()) + "(" +
-                       ins->get_literal().to_string() + ")";
+            if(ins->name() == "@literal"){
+                auto string_literal = ins->get_literal().to_string();
+
+                if (string_literal == "inf") {
```

Comment:
We shouldn't rely on the string representation which could change, we should just check with `std::isinf`.

Reply:
Done